pkg/controller/controllercmd/cmd.go

```diff
@@ -146,10 +153,23 @@ func (c *ControllerCommandConfig) StartController(ctx context.Context) error {
 			config.ServingInfo.CertFile = filepath.Join(certDir, "tls.crt")
 			config.ServingInfo.KeyFile = filepath.Join(certDir, "tls.key")
 			// nothing can trust this, so we don't really care about hostnames
-			_, err = ca.MakeAndWriteServerCert(config.ServingInfo.CertFile, config.ServingInfo.KeyFile, sets.NewString("localhost"), 30)
+			servingCert, err := ca.MakeServerCert(sets.NewString("localhost"), 30)
 			if err != nil {
 				return err
 			}
+			if err := servingCert.WriteCertConfigFile(config.ServingInfo.CertFile, config.ServingInfo.KeyFile); err != nil {
+				return err
+			}
+			crtContent := &bytes.Buffer{}
+			keyContent := &bytes.Buffer{}
+			if err := servingCert.WriteCertConfig(crtContent, keyContent); err != nil {
+				return err
+			}
+
+			// If we generate our own certificates, then we want to specify empty content to avoid a starting race.  This way,
```

Comment:
comment says "specify empty content" but this doesn't look like empty content, what am i missing?

Reply:
> comment says "specify empty content" but this doesn't look like empty content, what am i missing?

stale comments.  clearly I should not write comments :)